app/src/test/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreenLogicTest.kt

```diff
@@ -84,10 +84,23 @@ class NewMedicalHistoryScreenLogicTest {
   fun `when save is clicked with selected answers then patient with the answers should be saved and summary screen should be opened`() {
     // given
     val savedPatient = TestData.patient(uuid = patientUuid)
-    whenever(patientRepository.saveOngoingEntryAsPatient(eq(user), eq(facility), any(), any(), any())).thenReturn(Single.just(savedPatient))
+    val addressUuid = UUID.fromString("79f6d1bc-0e10-480b-9cce-43c628d827b7")
+
+    val uuidGenerator = mock<UuidGenerator>()
+    whenever(uuidGenerator.v4()).thenReturn(patientUuid, addressUuid, medicalHistoryUuid)
```

Comment:
In the sense? We are already using explicit IDs here since `patientUuid`, ``addressUuid`, `medicalHistoryUuid` are all fixed IDs, right?

Reply:
My doubt was mainly around why we were using `whenever` to return the UUID, I thought we could use `FakeUuidGenerator` directly?

Reply:
We cannot since it only allows returning one UUID. Here, what we need is it to return these specific UUIDs in order whenever the `.v4()` method is called since the method is invoked multiple times in the same test.

This will be fixed when we have different methods for generating IDs on a resource level.